SalesforceFoundation/EDA
# [EDA] Multiple contacts marked as deceased in the same bulk operation are not properly handled

# Critical Changes

# Changes
## Updates for Account Naming
- We've updated the way EDA handles changes to multiple Contact records from different Accounts. Now, when multiple Contacts in an Account are updated in different ways (for example, one might be specified as Deceased, another might have an updated name), all the Account names are properly updated.  In cases when an Account has only one remaining "active" Contact (the last Contact who isn't specified as Deceased or Excluded from Household Naming) and that Contact is deleted, merged, or moved to a different account, the Account name is NOT updated.

# Issues Closed
Closes #1050
# New Metadata

# Deleted Metadata

# Testing Notes
Scenario 1: Bulk test: Update x amount of Contacts from different Accounts and mark the "Deceased" or "Exclude From Household Naming" checkboxes true. 
Steps: 
- Create multiple Contacts and Accounts
- Ensure you have multiple Contacts on the same Account
- Ensure you have multiple Accounts
- Update 1 Contact from each Account by marking either "Deceased" or "Exclude From Household Naming". 
- All Account names should be updated with just the Contacts left on the Account that are not marked Deceased or Excluded from Household Naming. 

Scenario 2: Bulk test: Update x amount of Contacts from different Accounts and mark some of the Contacts "Deceased" or "Exclude From Household Naming" checkboxes true, and some Contacts update their FirstName or LastName but do not mark them as "Deceased" or "Exclude From Household Naming"
Steps: 
- Create multiple Contacts and Accounts
- Ensure you have multiple Contacts on the same Account
- Ensure you have multiple Accounts
- Update 1 Contact from each Account by marking either "Deceased" or "Exclude From Household Naming" or updating their FirstName or LastName. 
- All Account names should be updated with just the Contacts left on the Account that are not marked "Deceased" or "Excluded from Household Naming" and the Contacts that were updated but nor marked "Deceased" or  "Excluded from Household Naming" should still reflect on the Account name. 

Scenario 3: Non bulk: Update 1 Contact from an Account and marked it "Deceased" or "Exclude From Household Naming". 
Steps:
- Create an Account with multiple Contacts
- Update 1 Contact and marked "Deceased" or "Exclude From Household Naming"
- Account name should be updated with just the Contacts left on the Account that are not marked "Deceased" or "Exclude From Household Naming"
